packages/pwa-kit-react-sdk/src/ssr/universal/components/route-component/index.js

```diff
@@ -402,17 +402,17 @@ export const routeComponent = (Wrapped, isPage, locals) => {
  *
  * @private
  */
-export const getRoutes = (locals = {}) => {
+export const getRoutes = async (locals = {}, req = {}) => {
     let _routes = routes
     const {applicationExtensions = []} = locals
     if (typeof routes === 'function') {
-        _routes = routes()
+        _routes = await routes(locals)
     }
 
     // Call the `extendRoutes` function for all the Application Extensions.
-    applicationExtensions.forEach((applicationExtension) => {
-        _routes = applicationExtension.extendRoutes(_routes)
-    })
+    for (const applicationExtension of applicationExtensions) {
+        _routes = await applicationExtension.extendRoutes(_routes, req)
+    }
```

Comment:
Will this code run the promises in parallel or one after another?

Reply:
This code will run one after another